components/api/core/api_impl.bal

```diff
@@ -820,3 +820,183 @@ int resultLimit) returns http:Response {
         return buildUnknownErrorResponse();
     }
 }
+
+# Delete an artifact
+#
+# + deleteArtifactReq - received request which contains header
+# + orgName - organization name that the artifact is belong to
+# + imageName - image name of the artifact
+# + artifactVersion - version of the artifact
+# + return - http resonce (200 if success, 401, 403 or 500 otherwise)
+public function deleteArtifact(http:Request deleteArtifactReq, string orgName, string imageName, string artifactVersion) returns http:Response {
+    if (deleteArtifactReq.hasHeader(constants:AUTHENTICATED_USER)) {
+        string userId = deleteArtifactReq.getHeader(constants:AUTHENTICATED_USER);
+        log:printInfo(io:sprintf("User \'%s\' is attempting to delete the artifact \'%s/%s:%s\'", userId, orgName, imageName, artifactVersion));
+        http:Response resp;
+
+        transaction {
+            var transactionId = transactions:getCurrentTransactionId();
+            log:printDebug(io:sprintf("Started transaction \'%s\' for deleting the artifact \'%s/%s:%s\'", transactionId, orgName,
+            imageName, artifactVersion));
+            int | error? deletedRowCount = db:deleteArtifactFromDb(userId, orgName, imageName, artifactVersion);
+            if (deletedRowCount is int) {
+                if (deletedRowCount == 1) {
+                    var deleteResult = deleteArtifactFromRegistry(deleteArtifactReq, orgName, imageName, artifactVersion);
+                    if (deleteResult is error) {
+                        log:printError(io:sprintf("Failed to delete the artifact \'%s/%s:%s\'", orgName, imageName, artifactVersion),
```

Comment:
Please add failed to delete artifact from Docker registry to this log to add more context.